frontend/src/features/admin-form/create/builder-and-design/BuilderAndDesignDrawer/FieldListDrawer/FieldListOption.tsx

```diff
@@ -62,6 +69,21 @@ export const FieldListOption = forwardRef<FieldOptionProps, 'button'>(
       () => BASICFIELD_TO_DRAWER_META[fieldType],
       [fieldType],
     )
+    const { data: form } = useCreateTabForm()
+    const numFields = useMemo(() => form?.form_fields?.length ?? 0, [form])
```

Comment:
Just checking: is doing memoization based on `form` sufficient? As in: could `form` as a reference be constant, but `form.form_fields.length` have changed?

(Sorry, I know we use `useMemo()` everywhere, but this still looks a little funny to me 😅)

Reply:
I can't find documentation on this anywhere, but I'm pretty sure `react-query` changes the data reference when the data changes - the alternative doesn't make any sense! the only situation when the reference remains the same should be when the data hasn't change, due to [structural sharing](https://react-query.tanstack.com/guides/important-defaults#_top):

> Query results by default are structurally shared to detect if data has actually changed and if not, the data reference remains unchanged to better help with value stabilization with regards to useMemo and useCallback. If this concept sounds foreign, then don't worry about it! 99.9% of the time you will not need to disable this and it makes your app more performant at zero cost to you.